library/core/src/slice/mod.rs

```diff
@@ -3594,6 +3602,166 @@ impl<T> [T] {
         }
     }
 
+    /// Get a subslice where the first element and the element one past the end is aligned
```

Comment:
This is a bit confusing as it indicates that there might be an element after the end.

I think it'd be better to talk about the first element being aligned and the size of the return slice being a multiple of the alignment.